great_expectations/data_context/data_context.py

```diff
@@ -867,49 +858,26 @@ def save_config_variable(self, config_variable_name, value):
         with open(config_variables_filepath, "w") as config_variables_file:
             yaml.dump(config_variables, config_variables_file)
 
-    # TODO: deprecate
-    def delete_datasource(self, datasource_name=None):
-        """Delete a data source
-        Args:
-            datasource_name: The name of the datasource to delete.
-
-        Raises:
-            ValueError: If the datasource name isn't provided or cannot be found.
-        """
-        if datasource_name is None:
-            raise ValueError("Datasource names must be a datasource name")
-        else:
-            datasource = self.get_datasource(datasource_name)
-            if datasource:
-                # delete datasources project config
-                del self._project_config_with_variables_substituted.datasources[
-                    datasource_name
-                ]
-                del self._project_config.datasources[datasource_name]
-                del self._cached_datasources[datasource_name]
-            else:
-                raise ValueError("Datasource {} not found".format(datasource_name))
-
-    def delete_datasource(self, datasource_name=None):
+    def delete_datasource(self, name):
```

Comment:
I like the new name, but changing this parameter name could break user's code. If it's important to change, we should deprecate and support the old name for a while.

That said, we're not very consistent I think wrt "datasource_name" vs "name" (e.g. batch_request). I suspect the theory that "name" is used when the name "datasource" is in the method name, so there is no ambiguity? If so, let's write that into the style guide and do a comprehensive review (i.e. out of scope for this PR).

Reply:
@jcampbell I changed it to be consistent with most other cases where we use `name` in the methods, such as `add()`, `get()`, `delete()`, etc.  However, you are making a good point in that it may break the user code, so I am changing it back.  Perhaps we can make a more standards-aimed change in all key classes at some point in the future.  Thank you.